net-setup.sh

```diff
@@ -0,0 +1,135 @@
+#!/bin/sh
+#
+# Copyright (c) 2018 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+usage() {
+    cat <<EOF
+$0 [--config <file>] [--iface <interface>] [start|up] [stop|down]
+               [any optional parameters for ip command]"
+
+If no parameters are given, then "zeth" network interface and "zeth.conf"
+configuration file are used. The script waits until user presses CTRL-c
+and then removes the network interface.
+
+Examples:
+
+$ net-setup.sh
+$ net-setup.sh --config zeth-vlan.conf
+$ net-setup.sh --config my-own-config.conf --iface foobar
+
+It is also possible to let the script return and then stop the network
+interface later. Is can be done by first creating the interface with
+"start" or "up" command, and then later remove the interface with
+"stop" or "down" command.
+
+$ net-setup.sh start
+do your things here
+$ net-setup.sh stop
+
+$ net-setup.sh --config my-own-config.conf up
+do your things here
+$ net-setup.sh --config my-own-config.conf down
+
+Any extra parameters that the script does not know, are passed directly
+to "ip" command.
+
+$ net-setup.sh --config my-own-config.conf --iface foo user bar
+
+EOF
+    exit
+}
+
+if [ "$1" = "-h" -o "$1" = "--help" ]; then
+    usage
+fi
+
+if [ `id -u` != 0 ]; then
+    echo "Run this script as a root user!"
+    sudo $0 $@
+    exit
+fi
+
+IFACE=zeth
+
+# Default config file setups default connectivity IP addresses
+CONF_FILE=./zeth.conf
+
+while [ $# -gt 0 ]
+do
+    case $1 in
+	--config)
+	    CONF_FILE="$2"
+	    shift 2
+	    ;;
+	--iface)
+	    IFACE="$2"
+	    shift 2
+	    ;;
+	--help|-h)
+	    usage
+	    ;;
+	up|start)
+	    ACTION=start
+	    shift
+	    ;;
+	down|stop)
+	    ACTION=stop
+	    shift
+	    ;;
+	*)
+	    break
+	    ;;
+    esac
+done
+
+if [ ! -f "$CONF_FILE" ]; then
+    if [ ! -f "${0%/*}/$CONF_FILE" ];then
+	echo "No such file '$CONF_FILE'"
+	exit
+    fi
+
+    CONF_FILE="${0%/*}/$CONF_FILE"
+fi
+
+echo "Using $CONF_FILE configuration file."
+
+STOPPED=0
+trap ctrl_c INT TERM
+
+ctrl_c() {
+    STOPPED=1
+}
+
+if [ "$ACTION" != stop ]; then
+    echo "Creating $IFACE"
+    ip tuntap add $IFACE mode tap $@
+
+    # The idea is that the configuration file will setup
+    # the IP addresses etc. for the created interface.
+    . "$CONF_FILE" $IFACE
```

Comment:
if conf file supposed to be sourced or executed? I'm running into some issues:

```
Creating zeth
+ ip tuntap add zeth mode tap
+ . ./zeth.conf zeth
+ INTERFACE=
+ HWADDR=00:00:5e:00:53:ff
+ IPV6_ADDR_1=2001:db8::2
+ IPV6_ROUTE_1=2001:db8::/64
+ IPV4_ADDR_1=192.0.2.2
+ IPV4_ROUTE_1=192.0.2.0/24
+ ip link set dev up
Cannot find device "up"

```

It seems the conf file does not receive the argument if sourced.

Reply:
Hmm, interesting. The idea was that the conf file is sourced and it works for me. How do you start the script, could you give the full command line?

Reply:
```
$ sudo ./net-setup.sh 
+ [  = -h -o  = --help ]
+ id -u
+ [ 0 != 0 ]
+ IFACE=zeth
+ CONF_FILE=./zeth.conf
+ [ 0 -gt 0 ]
+ [ ! -f ./zeth.conf ]
+ echo Using ./zeth.conf configuration file.
Using ./zeth.conf configuration file.
+ STOPPED=0
+ trap ctrl_c INT TERM
+ [  != stop ]
+ echo Creating zeth
Creating zeth
+ ip tuntap add zeth mode tap
+ . ./zeth.conf zeth
+ set -x
+ INTERFACE=
+ HWADDR=00:00:5e:00:53:ff
+ IPV6_ADDR_1=2001:db8::2
+ IPV6_ROUTE_1=2001:db8::/64
+ IPV4_ADDR_1=192.0.2.2
+ IPV4_ROUTE_1=192.0.2.0/24
+ ip link set dev up
Cannot find device "up"
+ ip link set dev address 00:00:5e:00:53:ff
Error: either "dev" is duplicate, or "00:00:5e:00:53:ff" is a garbage.
+ ip -6 address add 2001:db8::2 dev
Command line is not complete. Try option "help"
+ ip -6 route add 2001:db8::/64 dev
Command line is not complete. Try option "help"
+ ip address add 192.0.2.2 dev
Command line is not complete. Try option "help"
+ ip route add 192.0.2.0/24 dev
Command line is not complete. Try option "help"
+ [  =  ]
+ [ 0 -eq 0 ]
+ sleep 1d

```

Reply:
I get the same issue if running the script with "up" command

Reply:
It seems to be shell issue, I have Debian testing where sh is dash

Reply:
Yep, I will set the script to use bash instead.